thoth/adviser/steps/tensorflow_avx2.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+# thoth-adviser
+# Copyright(C) 2020 Fridolin Pokorny
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Recommend TensorFlow builds optimized for AVX2 enabled CPU processors."""
+
+from typing import Any
+from typing import Optional
+from typing import Tuple
+from typing import List
+from typing import Dict
+from typing import TYPE_CHECKING
+import logging
+
+from thoth.python import PackageVersion
+
+from ..enums import RecommendationType
+from ..state import State
+from ..step import Step
+
+
+if TYPE_CHECKING:
+    from ..pipeline_builder import PipelineBuilderContext
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TensorFlowAVX2Step(Step):
+    """A step that recommends AICoE TensorFlow builds optimized for AVX2 enabled CPU processors."""
+
+    _SCORE_ADDITION = 0.2
+    _JUSTIFICATION_ADDITION = [
+        {
+            "type": "INFO",
+            "message": "AICoE TensorFlow builds are optimized for AVX2 instruction sets that your CPU supports",
+        }
+    ]
+
+    # A tuple (CPU_FAMILY, CPU_MODEL) of Intel processors supporting AVX2:
+    #   https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#CPUs_with_AVX2
+    #   https://en.wikichip.org/wiki/intel/cpuid
+    _AVX2_CPUS = frozenset(
```

Comment:
I'm not 100% sure with this listing - not sure if all the CPUs are captured correctly.

Reply:
Please create an issue so that we don’t lose it, let’s check again later

Reply:
@sub-mod

Reply:
The issue opened #1022